src/ripple/app/main/Application.cpp

```diff
@@ -368,7 +370,10 @@ class ApplicationImp
     std::vector <std::unique_ptr<Stoppable>> websocketServers_;
 
     boost::asio::signal_set m_signals;
-    beast::WaitableEvent m_stop;
+
+    std::condition_variable cv_;
```

Comment:
Random thought: a single `std::experimental::latch` would replace the `condition_variable`, the `mutex` and the `bool` and it would be much more concise and expressive. I wonder if that's still part of the Concurrency TS.

Reply:
It looks like `latch` is being actively proposed:
http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0666r2.pdf
But I don't see it in the latest draft.  It was not part of the full committee motions at the last meeting, so it is probably still working its way through the Concurrency subgroup.